sretoolbox/utils/cache.py

```diff
@@ -0,0 +1,167 @@
+"""
+Caching helper decorator
+"""
+
+import functools
+import weakref
+
+_CACHES = weakref.WeakKeyDictionary()
```

Comment:
I can't think on a case when the child dictionay will be garbage collected. Once you decora a function, the reference to that dict will last the entyre session. What am I missing?

Reply:
The class instance using @cache.* decorators on its methods is used as a key to this WeakKeyDictionary and the value is its own scoped cache dict.

The cache dict will get garbage collected when the class instance is gone.

Reply:
Right! That makes sense and I like it :)
But then we are broken when we decorate a function (not a method in a class).

Reply:
I see... very very good catch! I'll try to come up with a concept that works without the need for a class instance.